gravityflow/gravityflow
# Fix partial entry data loss on update and submission and update setting for GF 2.5

## Description
re: [HS#17001](https://secure.helpscout.net/conversation/1480918751/17001?folderId=1113535)

This updates the workflow setting we add to the Partial Entries feed configuration page for compatibility with Gravity Forms 2.5. At the moment the setting remains hidden when enabling PE until the feed is saved and the page reloads. The setting, a toggle, will be displayed live for 2.5 without the page reload.

It also fixes an issue where values added to the partial entry for administrative fields by the Update Fields step (Form Connector extension) are lost when the partial entry is updated by the PE add-on.

You'll need Partial Entries v1.6.1.

## Testing instructions
- Create a form with several fields of different types e.g. text, name, address
- Set all fields except a text field as administrative
- Add an Update Fields step which is configured to update the administrative fields of the current form and entry with custom values
- Add an approval or partial entry submission step, the configuration doesn’t matter we just need it to pause the workflow
- Enable the PE add-on for the form + the workflow
- Preview form
- Add a value to the text field
- Watch the browsers network panel for ajax request to save the partial entry
- In another tab go and view the partial entry to confirm the Update Fields step has set the administrative field values
- In the preview tab change the text field value
- Wait for the ajax request to update the partial entry
- In the other tab reload the entry to confirm the text value updated and the administrative field values remained
- Check that the workflow is waiting on the second step
- In the preview tab change the text field value again and submit
- Check the entry to confirm the text value was updated and the administrative field values remained

## Automated Test Enhancements
N/A

## Documentation Changes
https://docs.gravityflow.io/article/258-partial-entries
The above page should be updated to explain that the values of non-administrative fields set by workfow steps will be lost when the partial entry is updated by the Partial Entries add-on when the heartbeat occurs, when the user pages through the form, or when the user submits the form.

## Checklist:
